packages/core/src/components/dialog/_dialog.scss

```diff
@@ -97,6 +97,7 @@ $dialog-header-padding: $pt-spacing;
   .#{$ns}-heading {
     @include overflow-ellipsis;
     flex: 1 1 auto;
+    font-size: 14px;
```

Comment:
Not sure if there's a special variable I can use here to reference the font size of `h4` titles.

Reply:
Sounds reasonable! Thanks for the discussion here :)